src/osip-simulation-view/src/main/java/edu/kit/pse/osip/simulation/view/control/AbstractTankTab.java

```diff
@@ -1,14 +1,163 @@
+
+        temperatureSlider = new Slider(min,
+                SimulationConstants.MAX_TEMPERATURE - SimulationConstants.CELCIUS_OFFSET,
+                tank.getLiquid().getTemperature() - SimulationConstants.CELCIUS_OFFSET);
+        temperatureSlider.setShowTickLabels(true);
+        temperatureSlider.setShowTickMarks(true);
+        temperatureSlider.setMajorTickUnit(25);
+        temperatureSlider.setMinorTickCount(9);
+        temperatureSlider.setMinWidth(ViewConstants.CONTROL_SLIDER_WIDTH);
+        pane.add(temperatureSlider, 0, row);
+        GridPane.setMargin(temperatureSlider, ViewConstants.CONTROL_PADDING);
+
+        //Labels to show the current value and unit
+        TextField temperatureValue = new TextField("" + tank.getLiquid().getTemperature());
+        temperatureValue.setMaxWidth(ViewConstants.CONTROL_INPUT_WIDTH);
+        pane.add(temperatureValue, 1, row);
+        GridPane.setMargin(temperatureValue, ViewConstants.CONTROL_PADDING);
+        Label temperatureSign = new Label(t.getString("simulation.view.temperatureUnit"));
+        pane.add(temperatureSign, 2, row);
+        GridPane.setMargin(temperatureSign, ViewConstants.CONTROL_PADDING);
+
+        connectElements(temperatureSlider, temperatureValue, false);
+        row++;
+    }
+
+    private void createOutFlowSlider(GridPane pane, AbstractTank tank) {
+        Translator t = Translator.getInstance();
+
+        outFlowSlider = new Slider(0, 100, tank.getOutPipe().getValveThreshold());
+        outFlowSlider.setShowTickLabels(true);
+        outFlowSlider.setShowTickMarks(true);
+        outFlowSlider.setMajorTickUnit(10);
+        outFlowSlider.setMinorTickCount(9);
+        outFlowSlider.setSnapToTicks(true);
+        outFlowSlider.setMinWidth(ViewConstants.CONTROL_SLIDER_WIDTH);
+        pane.add(outFlowSlider, 0, row);
+        GridPane.setMargin(outFlowSlider, ViewConstants.CONTROL_PADDING);
+
+        //TextField and Label to show the current value and unit
+        TextField outFlowValue = new TextField("" + tank.getOutPipe().getValveThreshold());
+        outFlowValue.setMaxWidth(ViewConstants.CONTROL_INPUT_WIDTH);
+        pane.add(outFlowValue, 1, row);
+        GridPane.setMargin(outFlowValue, ViewConstants.CONTROL_PADDING);
+        Label outFlowSign = new Label(t.getString("simulation.view.flowUnit"));
+        pane.add(outFlowSign, 2, row);
+        GridPane.setMargin(outFlowSign, ViewConstants.CONTROL_PADDING);
+
+        //Listener to update the Label
+        connectElements(outFlowSlider, outFlowValue, true);
+        row++;
+    }
+
+    /**
+     * Connects a slider and a textfield so that if one is changed the other is changed to
+     * show the same value.
+     * @param slider The Slider to be connected
+     * @param textField The TextField to be connected
+     * @param isIntegerValue    If true all values are set to be integers
+     */
+    protected void connectElements(Slider slider, TextField textField, boolean isIntegerValue) {
```

Comment:
Da gibt es direkt in javafx eine bidirektionale Verbindung. Bitte Bordmittel nutzen. Für ein Beispiel zu Martin in die Settings schauen